pennylane/transforms/op_transforms.py

```diff
@@ -0,0 +1,481 @@
+
+        .. code-block:: python
+
+            @qml.op_transform
+            def simplify_rotation(op):
+                if op.name == "Rot":
+                    params = op.parameters
+                    wires = op.wires
+
+                    if qml.math.allclose(params, 0):
+                        return
+
+                    if qml.math.allclose(params[1:2], 0):
+                        return qml.RZ(params[0], wires)
+
+                return op
+
+            @simplify_rotation.tape_transform
+            @qml.qfunc_transform
+            def simplify_rotation(tape):
+                for op in tape.operations + tape.measurements:
+                    if op.name == "Rot":
+                        simplify_rotation(op)
+                    else:
+                        qml.apply(op)
+
+        We can now use this combined operator and quantum function transform in compilation pipelines:
+
+        .. code-block:: python
+
+            @qml.qnode(dev)
+            @qml.compile(pipeline=[simplify_rotation])
+            def circuit(weights):
+                ansatz(weights)
+                qml.CNOT(wires=[0, 1])
+                qml.Rot(0.0, 0.0, 0.0, wires=0)
+                return qml.expval(qml.PauliX(1))
+        """
+        self._tape_fn = fn
+        return self
+
+    def _create_wrapper(self, obj, *targs, wire_order=None, **tkwargs):
+
+        if isinstance(obj, qml.operation.Operator):
+            # Input is a single operation.
+            # op_transform(obj, *transform_args)
+            if wire_order is not None:
+                tkwargs["wire_order"] = wire_order
+
+            wrapper = self.fn(obj, *targs, **tkwargs)
+
+        elif isinstance(obj, qml.tape.QuantumTape):
+            # Input is a quantum tape. Get the quantum tape.
+            tape, verified_wire_order = _make_tape(obj, wire_order)
+
+            if wire_order is not None:
+                tkwargs["wire_order"] = verified_wire_order
+
+            wrapper = self.tape_fn(tape, *targs, **tkwargs)
+
+        elif callable(obj):
+            # Input is a QNode, or qfunc. Get the quantum tape.
+            def wrapper(*args, **kwargs):
+                nonlocal wire_order
+                tape, verified_wire_order = _make_tape(obj, wire_order, *args, **kwargs)
+
+                if wire_order is not None or (
+                    "wire_order" in self._sig and isinstance(obj, qml.QNode)
+                ):
+                    # Use the verified wire order if:
+                    # - wire_order was passed to the transform
+                    # - The object is a QNode, and the function takes a wire_order argument
+                    tkwargs["wire_order"] = verified_wire_order
+
+                if isinstance(tape, qml.operation.Operator):
+                    return self.fn(tape, *targs, **tkwargs)
+
+                if self.is_qfunc_transform:
+                    return self.tape_fn(obj, *kwargs, **tkwargs)(*args, **kwargs)

```

Comment:
I'm confused here:
* How come we use `kwargs` as the positional arguments passed to `tape_fn`;
* Do we evaluate the result of `self.tape_fn(obj, *kwargs, **tkwargs)` because we were using the qfunc transform and we need to evaluate that too?

Reply:
you are exactly right!
```suggestion
                if self.is_qfunc_transform:
                    # we must evaluate the qfunc transform at the original
                    # function arguments
                    return self.tape_fn(obj, *kwargs, **tkwargs)(*args, **kwargs)
```